test/unit/app/repository/application-repository.test.js

```diff
@@ -55,12 +62,118 @@ describe('Application Repository test', () => {
+                model: data.models.status,
+                attributes: ['status']
+              }],
+            where: {
+              'data.organisation.sbi': searchText
+            }
+          })
+
+          expect(data.models.application.count).toHaveBeenCalledWith({
+            where: {
+              'data.organisation.sbi': searchText
+            }
+          })
+          break
+        case 'ref':
+          expect(data.models.application.findAll).toHaveBeenCalledWith({
+            order: [['createdAt', 'DESC']],
+            limit,
+            offset,
+            include: [
+              {
+                model: data.models.status,
+                attributes: ['status']
+              }],
+            where: {
+              reference: searchText
+            }
+          })
+
+          expect(data.models.application.count).toHaveBeenCalledWith({
+            where: {
+              reference: searchText
+            }
+          })
+          break
+        case 'status':
+          expect(data.models.application.findAll).toHaveBeenCalledWith({
+            order: [['createdAt', 'DESC']],
+            limit,
+            offset,
+            include: [
+              {
+                model: data.models.status,
+                attributes: ['status'],
+                where: { status: undefined }
+              }]
+          })
+
+          expect(data.models.application.count).toHaveBeenCalledWith({
+            include: [
+              {
+                model: data.models.status,
+                attributes: ['status'],
+                where: { status: undefined }
+              }]
+          })
+          break
+      }
+    } else {
+      expect(data.models.application.findAll).toHaveBeenCalledWith({
+        order: [['createdAt', 'DESC']],
+        limit,
+        offset,
+        include: [
+          {
+            model: data.models.status,
+            attributes: ['status']
+          }]
       })
     }
   })
+  test.each([
+    { limit, offset, searchText: undefined, searchType: 'sbi' },
+    { limit, offset, searchText: '333333333', searchType: 'sbi' },
+    { limit, offset, searchText: '', searchType: 'sbi' },
+    { limit, offset, searchText: undefined, searchType: 'ref' },
+    { limit, offset, searchText: '333333333', searchType: 'ref' },
+    { limit, offset, searchText: '', searchType: 'ref' },
+    { limit, offset, searchText: undefined, searchType: 'status' },
+    { limit, offset, searchText: '333333333', searchType: 'status' },
+    { limit, offset, searchText: '', searchType: 'status' }
+  ])('getApplications returns empty array when no application found.', async ({ searchText, searchType, limit, offset }) => {
+    data.models.application.count.mockReturnValue(0)
+    await repository.searchApplications(searchText, searchType, offset, limit)
 
+    expect(data.models.application.count).toHaveBeenCalledTimes(1)
+    expect(data.models.application.findAll).not.toHaveBeenCalled()
```

Comment:
It would be good to see an expectation on what `count` was called with then should the call change the test would catch it.